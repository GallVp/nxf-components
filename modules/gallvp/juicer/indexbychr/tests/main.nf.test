nextflow_process {

    name "Test Process JUICER_INDEXBYCHR"
    script "../main.nf"
    config "./nextflow.config"
    process "JUICER_INDEXBYCHR"

    tag "modules"
    tag "modules_gallvp"
    tag "juicer"
    tag "juicer/indexbychr"
    tag "yahs/juicerpre"
    tag "sort"

    test("hongyang - hic") {

        setup {
            run("YAHS_JUICERPRE") {
                script "../../../yahs/juicerpre/main.nf"
                process {
                    """
                    input[0] = [
                        [ id:'test', single_end:false ], // meta map
                        file( 'https://github.com/Plant-Food-Research-Open/assemblyqc/raw/4f9fba392cf7212d61c6245fd72339e09164ce44/tests/hic/testdata/HYh1h2_Chr01_6000000_6015000.bam' , checkIfExists: true)
                    ]

                    input[1] = file( 'https://github.com/Plant-Food-Research-Open/assemblyqc/raw/4f9fba392cf7212d61c6245fd72339e09164ce44/tests/hic/testdata/HYh1h2_Chr01_6000000_6015000.agp' , checkIfExists: true)
                    input[2] = file( 'https://github.com/Plant-Food-Research-Open/assemblyqc/raw/4f9fba392cf7212d61c6245fd72339e09164ce44/tests/hic/testdata/HYh1h2_Chr01_6000000_6015000.fasta.fai' , checkIfExists: true)
                    """
                }
            }

            run("SORT") {
                script "../../../sort/main.nf"

                process {
                    """
                    input[0] = YAHS_JUICERPRE.out.txt
                    """
                }
            }
        }

        when {
            params {
                sort = '-k2,2d -k6,6d'
                yahs_juicerpre = ''
            }
            process {
                """
                input[0] = SORT.out.sorted
                input[1] = 500000
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("hongyang - hic - mode - assembly") {

        setup {
            run("YAHS_JUICERPRE") {
                script "../../../yahs/juicerpre/main.nf"
                process {
                    """
                    input[0] = [
                        [ id:'test', single_end:false ], // meta map
                        file( 'https://github.com/Plant-Food-Research-Open/assemblyqc/raw/4f9fba392cf7212d61c6245fd72339e09164ce44/tests/hic/testdata/HYh1h2_Chr01_6000000_6015000.bam' , checkIfExists: true)
                    ]

                    input[1] = file( 'https://github.com/Plant-Food-Research-Open/assemblyqc/raw/4f9fba392cf7212d61c6245fd72339e09164ce44/tests/hic/testdata/HYh1h2_Chr01_6000000_6015000.agp' , checkIfExists: true)
                    input[2] = file( 'https://github.com/Plant-Food-Research-Open/assemblyqc/raw/4f9fba392cf7212d61c6245fd72339e09164ce44/tests/hic/testdata/HYh1h2_Chr01_6000000_6015000.fasta.fai' , checkIfExists: true)
                    """
                }
            }
        }

        when {
            params {
                sort = ''
                yahs_juicerpre = '-a'
            }
            process {
                """
                input[0] = YAHS_JUICERPRE.out.txt
                input[1] = 500000
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("hongyang - hic - stub") {

        options '-stub'

        setup {
            run("YAHS_JUICERPRE") {
                script "../../../yahs/juicerpre/main.nf"
                process {
                    """
                    input[0] = [
                        [ id:'test', single_end:false ], // meta map
                        file( 'https://github.com/Plant-Food-Research-Open/assemblyqc/raw/4f9fba392cf7212d61c6245fd72339e09164ce44/tests/hic/testdata/HYh1h2_Chr01_6000000_6015000.bam' , checkIfExists: true)
                    ]

                    input[1] = file( 'https://github.com/Plant-Food-Research-Open/assemblyqc/raw/4f9fba392cf7212d61c6245fd72339e09164ce44/tests/hic/testdata/HYh1h2_Chr01_6000000_6015000.agp' , checkIfExists: true)
                    input[2] = file( 'https://github.com/Plant-Food-Research-Open/assemblyqc/raw/4f9fba392cf7212d61c6245fd72339e09164ce44/tests/hic/testdata/HYh1h2_Chr01_6000000_6015000.fasta.fai' , checkIfExists: true)
                    """
                }
            }

            run("SORT") {
                script "../../../sort/main.nf"

                process {
                    """
                    input[0] = YAHS_JUICERPRE.out.txt
                    """
                }
            }
        }

        when {

            params {
                sort = '-k2,2d -k6,6d'
                yahs_juicerpre = ''
            }

            process {
                """
                input[0] = SORT.out.sorted
                input[1] = 500000
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

}