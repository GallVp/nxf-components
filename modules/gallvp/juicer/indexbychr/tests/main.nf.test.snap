{
    "hongyang - hic - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.index.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,a560a217944c78776df7eefddbf13229"
                ],
                "index": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.index.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,a560a217944c78776df7eefddbf13229"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.6"
        },
        "timestamp": "2025-08-07T15:55:34.880328"
    },
    "hongyang - hic - mode - assembly": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.index.txt:md5,b0cdf09d4dc2d0b7254952842d987293"
                    ]
                ],
                "1": [
                    "versions.yml:md5,a560a217944c78776df7eefddbf13229"
                ],
                "index": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.index.txt:md5,b0cdf09d4dc2d0b7254952842d987293"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,a560a217944c78776df7eefddbf13229"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.6"
        },
        "timestamp": "2025-08-07T15:55:27.822324"
    },
    "hongyang - hic": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.index.txt:md5,bbffcc7d877fc43958000d8a431a7e00"
                    ]
                ],
                "1": [
                    "versions.yml:md5,a560a217944c78776df7eefddbf13229"
                ],
                "index": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.index.txt:md5,bbffcc7d877fc43958000d8a431a7e00"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,a560a217944c78776df7eefddbf13229"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.6"
        },
        "timestamp": "2025-08-07T15:55:22.387455"
    }
}